React/CSS-in-JS/Chakra.md

```diff
@@ -0,0 +1,629 @@
+
+Theming with Chakra UI is based on the [Styled System Theme Specification](https://system-ui.com/theme/)
+
+Example:
+
+```ts
+const overrides = {
+  colors: {
+    primary: {
+      100: "#ff0000",
+      80: "#ff1a1a",
+    },
+  },
+};
+```
+
+Except objects, theme override can take a function:
+
+```ts
+const overrides = {
+  colors: (props) => {
+    const { colorMode } = props;
+
+    return {
+      primary: {
+        100: colorMode === "dark" ? "#ff0000" : "#ffffff",
+        80: "#ff1a1a",
+      },
+    };
+  },
+};
+```
+
+Then in `props` we have access button props, `theme` object and `colorMode`.
+
+#### `chakra-ui/theme-tools`
+
+Chakra has a whole pallet of useful helpers:
+
+- `mode(lightMode, darkMode)(props)` function is the same as `colorMode === "dark" ? darkMode : lightMode`.
+- `orient` define `vertical` and `horizontal` style for component: `<Divider orientation="horizontal | vertical" />`
+- `transparentize` helper to make a color transparent `transparentize('blue.100', 0.3)`
+- `darken` darken a specified color `darken('blue.100', 0.5)` (there is also `lighten`, `blacken`, `whiten`)
+
+For more about `theme-tools` check out [github](https://github.com/chakra-ui/chakra-ui/tree/develop/packages/theme-tools/src)
+
+Theme folder structure:
+
+```bash
+src
+├── styles
+    └── theme
+        ├── index.ts # main theme endpoint
+        ├── styles.ts # global styles
+        ├── foundations # colors, typography, sizes...
+            ├── fontSizes.ts
+            └── colors.ts
+        ├── components # components styles
+            └── button.ts
+```
+
+<!-- TODO maybe detail description of each folder or list of files per folder? -->
+
+### Colors
+
+Color naming should a single value or an object with keys in range from 50 to 900.
+
+Example:
+
+```ts
+export const colors = {
+  white: "#ffffff",
+  primary: {
+    50: "#f6faff",
+    100: "#d7e9fd",
+    200: "#c0dcfc",
+    300: "#a7cefb",
+    400: "#8abdfa",
+    500: "#68aaf8",
+    600: "#3d92f6",
+    700: "#0070f3",
+    800: "#0064d8",
+    900: "#0055b9",
+  },
+};
+```
+
+### Pseudo props
+
+Pseudo props in ChakraUI can be passed as props to component. Full list of props can be found in [docs](https://chakra-ui.com/docs/features/style-props#pseudo)
+
+Example:
+
+```tsx
+<Button
+  bg="blue.100"
+  _hover={{
+    background: "white",
+  }}
+>
+  Submit
+</Button>
+```
+
+Also pseudo elements can be used in theme or in `sx` prop
+
+```tsx
+<Button
+  sx={{
+    bg: 'blue.100',
+    _hover: {
+      bg: 'white
+    }
+  }}
+>
+  Submit
+</Button>
+
+// or
+
+<Button
+  sx={{
+    bg: 'blue.100',
+    ':hover': {
+      bg: 'white
+    }
+  }}
+>
+  Submit
+</Button>
+```
+
+### Global styles
+
+Global styles are theme-aware styles you can apply to any html element globally.
+
+They are defined in `src/styles/theme/styles.ts` file.
+
+<!-- TODO maybe what should be defined in global -->
+
+Example:
+
```

Comment:
Can we also mention `resetCss` prop on ChakraProvider?

https://chakra-ui.com/docs/migration#2-update-the-themeprovider

Reply:
I will add it